### PR TITLE
Me: Update ReauthRequired to use Redux analytics

### PR DESCRIPTION
**Pull request by tyxla**
This PR refactors `ReauthRequired` to use Redux analytics instead of `eventRecorder`.

This PR should not introduce any visual or behavioral changes.

Part of #20053.

To test:
* Checkout this branch
* Make sure your ga debug is enabled: `localStorage.debug = 'calypso:analytics:ga,calypso:analytics:utils'`
* Using Chrome, in the developer tools, go to Application > Cookies and delete the `twostep_auth` cookie for the `public-api.wordpress.com` domain.
* Go to http://calypso.localhost:3000/me/security/two-step or any other page that may require a fresh 2fa token and you'll be presented with the reauth dialog.
* Verify the right event is being dispatched when:
  * Clicking the **Not You? Sign Out** link.
  * Focusing the **Verification Code** field.
  * Toggling the **Remember for 30 days** checkbox.
  * Clicking the **Send SMS Code** button.
  * Clicking the **Verify** button (you need to have input your code first to enable this button).

### CODE REVIEW

File: client/me/reauth-required/index.jsx [hunk 1]
```diff
@@ -42,6 +43,29 @@ const ReauthRequired = createReactClass( {
 		};
 	},
 
+	getClickHandler( action, callback ) {
+		return () => {
```
[the remark this answers is not in the document]
Comment (tyxla) [reply client/me/reauth-required/index.jsx]: Nope, we can't do that, this is a `createReactClass` 😢

File: client/me/reauth-required/index.jsx [hunk 2]
```diff
@@ -42,6 +43,29 @@ const ReauthRequired = createReactClass( {
...
+			this.props.recordGoogleEvent( 'Me', 'Clicked on ' + action );
+
+			if ( callback ) {
+				callback();
+			}
+		};
+	},
+
+	getCheckboxHandler( checkboxName ) {
+		return event => {
```
[the remark this answers is not in the document]
Comment (tyxla) [reply client/me/reauth-required/index.jsx]: Nope, we can't do that, this is a `createReactClass` 😢

File: client/me/reauth-required/index.jsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (ockham) [reply client/me/reauth-required/index.jsx]: Hmm, sure that won't work?

[the remark this answers is not in the document]
Comment (tyxla) [reply client/me/reauth-required/index.jsx]: Ah, I think I misread it the first time - it actually might work that way. Not a big deal tho, right?

[the remark this answers is not in the document]
Comment (ockham) [reply client/me/reauth-required/index.jsx]: No, not big deal at all.

**Pull request merged**